Documentation: Kedro-Viz React component usage (#1954)
kedro-org/kedro-viz: README.npm.md

## Usage

To use Kedro-Viz as a standalone React component, you can follow the example below. Please note that Kedro-Viz does not support server-side rendering (SSR). If you are using Next.js or another SSR framework, you should be aware of this limitation.

```javascript
import KedroViz from '@quantumblack/kedro-viz';
import '@quantumblack/kedro-viz/lib/styles/styles.min.css';

const MyApp = () => <KedroViz data={json} />;
```

To use with NextJS:

```javascript
import '@quantumblack/kedro-viz/lib/styles/styles.min.css';
import dynamic from 'next/dynamic';

const NoSSRKedro = dynamic(() => import('@quantumblack/kedro-viz'), {
  ssr: false,
});

const MyApp = () => <NoSSRKedro data={json} />;
```

The JSON can be obtained by running:

```bash
kedro viz run --save-file=<path-to-save-kedro-viz-data-to-a-directory>
```

On successful execution of the command above, it will generate a folder at the specified directory, containing the following structure: 

```
<filename>/api/
├── main
├── nodes
│   ├── 23c94afb
│   ├── 28754fab
│   ├── 2ab3579f
│   ├── 329e963c
│   ├── 369acf98
│   └── ...
└── pipelines
    ├── __default__
    ├── data_processing
    ├── data_science
    └── ...
```

> Please use the `main` file as the input JSON for the `data` prop in your Kedro-Viz component.

Reviewer: ```suggestion
Use the `main` file as the input JSON for the `data` prop in your Kedro-Viz component.
```

> We also recommend wrapping the `Kedro-Viz` component with a parent HTML/JSX element that has a specified height (as seen in the above example) in order for Kedro-Viz to be styled properly.

Reviewer: I don't see the example where it's wrapped in a parent element. Am I missing it?

Author: Thanks for catching, yes its not wrapped.
Is it good idea to show wrap in below example?
```
import KedroViz from '@quantumblack/kedro-viz';
import '@quantumblack/kedro-viz/lib/styles/styles.min.css';

const MyApp = () => <KedroViz data={json} />; 
```
  
  with
  
```
import KedroViz from '@quantumblack/kedro-viz';
import '@quantumblack/kedro-viz/lib/styles/styles.min.css';

const MyApp = () => <div style={{height: `100vh`}}><KedroViz data={json} /></div>; 
```

or create a separate code snippets with below text.
"We also recommend wrapping the `Kedro-Viz` component with a parent HTML/JSX element that has a specified height (as seen in the above example) in order for Kedro-Viz to be styled properly."

Reviewer: I think the second code example is what should be shown in the docs. That's the recommended way of doing it, as far as I still understand it, right? If so, then show that.

[![Slack Organisation](https://img.shields.io/badge/slack-chat-blueviolet.svg?label=Kedro%20Slack&logo=slack)](https://slack.kedro.org)

## Introduction

Kedro-Viz is an interactive development tool for building data science pipelines with [Kedro](https://github.com/kedro-org/kedro). Kedro-Viz also allows users to view and compare different runs in the Kedro project.

## Features

- ✨ Complete visualisation of a Kedro project and its pipelines
- 🎨 Supports light & dark themes out of the box
- 🚀 Scales to big pipelines with hundreds of nodes
- 🔎 Highly interactive, filterable and searchable
- 🔬 Focus mode for modular pipeline visualisation
- 📊 Rich metadata side panel to display parameters, plots, etc.
- 📊 Supports all types of [Plotly charts](https://plotly.com/javascript/)
- ♻️ Autoreload on code change
- 🧪 Supports tracking and comparing runs in a Kedro project
- 🎩 Many more to come

## Installation

- As a standalone React component (for embedding Kedro-Viz in your web application).

  To install the standalone React component:

  ```bash
  npm install @quantumblack/kedro-viz
  ```

## Usage

> To use Kedro-Viz as a standalone React component, you can follow the example below. However, please note that Kedro-Viz does not support server-side rendering (SSR). If you're using Next.js or another SSR framework, you should be aware of this limitation.

Reviewer 2: ```suggestion
To use Kedro-Viz as a standalone React component, you can follow the example below. Please note that Kedro-Viz does not support server-side rendering (SSR). If you are using Next.js or another SSR framework, you should be aware of this limitation.
```

```javascript
import KedroViz from '@quantumblack/kedro-viz';
import '@quantumblack/kedro-viz/lib/styles/styles.min.css';

const MyApp = () => <KedroViz data={json} />;
```

To use with NextJS:

```javascript
import '@quantumblack/kedro-viz/lib/styles/styles.min.css';
import dynamic from 'next/dynamic';

const NoSSRKedro = dynamic(() => import('@quantumblack/kedro-viz'), {
  ssr: false,
});

const MyApp = () => <NoSSRKedro data={json} />;
```

The JSON can be obtained by running:

```bash

> kedro viz run --save-file=filename

Reviewer 2: ```suggestion
kedro viz run --save-file=<path-to-save-kedro-viz-data-to-a-directory>
```

```

> The command will generate a folder with the specified <filename>, containing the following structure: 

Reviewer 2: ```suggestion
On successful execution of the command above, it will generate a folder at the specified directory, containing the following structure: 
```

| `pipelineBtn` | boolean | true | If `false` pipelineBtn button at the top of the primaryToolbar will be hidden |
| `sidebar` | boolean | true | If `false` sidebar on left will be minimized |

### Note
When `preview` is `true`, `display` and `visible` props are ignored as all major components are hidden.

All components are annotated to understand their positions in the Kedro-Viz UI.

![Kedro-Viz component annotation](https://raw.githubusercontent.com/jitu5/kedro-viz/main/.github/img/kedro-viz-annotation.png)

## Standalone Example Repository

We have created a [kedro-viz-standalone](https://github.com/kedro-org/kedro-viz-standalone.git) repository to demonstrate how to use Kedro-Viz in standalone mode or embedded in a React application.

This repository provides a fully functional example of Kedro-Viz, showcasing how to integrate it into your projects. The example includes setup instructions and demonstrates various features of Kedro-Viz.

To get started, clone the repository, install the dependencies and run the example React application:

```
git clone https://github.com/kedro-org/kedro-viz-standalone.git
cd kedro-viz-standalone
npm install
npm start
```

## Feature Flags

> Kedro-Viz uses features flags to roll out some experimental features. The following flags are currently in use:

Reviewer 2: ```suggestion
Kedro-Viz uses feature flags to roll out some experimental features. We have the following flags -
```

[![Downloads](https://static.pepy.tech/badge/kedro-viz/week)](https://pepy.tech/project/kedro-viz)
[![npm version](https://img.shields.io/npm/v/@quantumblack/kedro-viz.svg?color=cc3534)](https://badge.fury.io/js/%40quantumblack%2Fkedro-viz)
[![License](https://img.shields.io/badge/license-Apache%202.0-3da639.svg)](https://opensource.org/licenses/Apache-2.0)
[![code style: prettier](https://img.shields.io/badge/code_style-prettier-ff69b4.svg)](https://github.com/prettier/prettier)
[![Slack Organisation](https://img.shields.io/badge/slack-chat-blueviolet.svg?label=Kedro%20Slack&logo=slack)](https://slack.kedro.org)

## Introduction

Kedro-Viz is an interactive development tool for building data science pipelines with [Kedro](https://github.com/kedro-org/kedro). Kedro-Viz also allows users to view and compare different runs in the Kedro project.

## Features

- ✨ Complete visualisation of a Kedro project and its pipelines
- 🎨 Supports light & dark themes out of the box
- 🚀 Scales to big pipelines with hundreds of nodes
- 🔎 Highly interactive, filterable and searchable
- 🔬 Focus mode for modular pipeline visualisation

> - 📊 Rich metadata side panel to display parameters, plots, etc.

Reviewer 3: ```suggestion
- 📊 Rich metadata side panel to display parameters, plots, and more.
```

- 📊 Supports all types of [Plotly charts](https://plotly.com/javascript/)
- ♻️ Autoreload on code change
- 🧪 Supports tracking and comparing runs in a Kedro project
- 🎩 Many more to come

## Installation

- As a standalone React component (for embedding Kedro-Viz in your web application).

  To install the standalone React component:

  ```bash
  npm install @quantumblack/kedro-viz
  ```

## Usage

To use Kedro-Viz as a standalone React component, you can follow the example below. Please note that Kedro-Viz does not support server-side rendering (SSR). If you are using Next.js or another SSR framework, you should be aware of this limitation.

```javascript
import KedroViz from '@quantumblack/kedro-viz';
import '@quantumblack/kedro-viz/lib/styles/styles.min.css';

> const MyApp = () => <div style={{height: `100vh`}}><KedroViz data={json} /></div>; 

Reviewer 3: here too maybe?

Author: @rashidakanchwala like this?
```
<div style={{height: `100vh`}}>
  <KedroViz
      data={json}
      options={
        display: {
          globalToolbar: true,
          sidebar: true,
          metadataPanel: true,
          miniMap: true,
          expandAllPipelines: false,
        }
        visible: {
          exportBtn: true,
          labelBtn: true,
          layerBtn: true,
          pipelineBtn: true,
          sidebar: true,        
        }
        theme: "dark",
        tag: {
          enabled: {companies: true}
        },
        nodeType: {
          disabled: {parameters: true}
        }
      }
  />
 </div>
```

```

To use with NextJS:

```javascript
import '@quantumblack/kedro-viz/lib/styles/styles.min.css';
import dynamic from 'next/dynamic';

const NoSSRKedro = dynamic(() => import('@quantumblack/kedro-viz'), {
  ssr: false,
});

const MyApp = () => <NoSSRKedro data={json} />;
```

The JSON can be obtained by running:

```bash
kedro viz run --save-file=<path-to-save-kedro-viz-data-to-a-directory>
```

On successful execution of the command above, it will generate a folder at the specified directory, containing the following structure: 

```
<filename>/api/
├── main
├── nodes
│   ├── 23c94afb
│   ├── 28754fab
│   ├── 2ab3579f
│   ├── 329e963c
│   ├── 369acf98
│   └── ...
└── pipelines
    ├── __default__
    ├── data_processing
    ├── data_science
    └── ...
```

Use the `main` file as the input JSON for the `data` prop in your Kedro-Viz component.

We also recommend wrapping the `Kedro-Viz` component with a parent HTML/JSX element that has a specified height (as seen in the above example) in order for Kedro-Viz to be styled properly.

**_Our documentation contains [additional examples on how to visualise with Kedro-Viz.](https://docs.kedro.org/en/stable/visualisation/index.html)_**

> ## Props
> 
> Below is the example with all possible props. 

Reviewer 3: ```suggestion
##  Configure Kedro-viz with `options`

The example below demonstrates how to configure your kedro-viz using different `options`.
```

```
<KedroViz
    data={json}
    options={
      display: {
        globalToolbar: true,
        sidebar: true,
        metadataPanel: true,
        miniMap: true,
        expandAllPipelines: false,
      }
      visible: {
        exportBtn: true,
        labelBtn: true,
        layerBtn: true,
        pipelineBtn: true,
        sidebar: true,        
      }
      theme: "dark",
      tag: {
        enabled: {companies: true}
      },
      nodeType: {
        disabled: {parameters: true}
      }
    }
 /> 
```

| Name         | Type    | Default | Description |
| ------------ | ------- | ------- | ----------- |

> | `data` | `{ edges: array (required), layers: array, nodes: array (required), tags: array }` | - | Pipeline data will be displayed on the chart |

Reviewer 3: ```suggestion
| `data` | `{ edges: array (required), layers: array, nodes: array (required), tags: array }` | - | Pipeline data tp be displayed on the chart |
```

> | options.theme | string | dark | select Kedro-Viz theme dark/light |

Reviewer 3: ```suggestion
| options.theme | string | dark | select `Kedro-Viz` theme : dark/light |
```

| options.tag | `{enabled: {<tagName>: boolean}}` | - | Configuration for tag options |
| options.nodeType | `{disabled: {parameters: boolean,task: boolean,data: boolean}}` | `{disabled: {parameters: true,task: false,data: false}}` | Configuration for node type options |
| options.display |  |  |  |

> | `globalToolbar` | boolean | true | If `false` Global sidebar on extreme left will be hidden |

Reviewer 3: nit - for description can we keep all of them simple. Like 

Show/Hide Global Toolbar 
Show/Hide Sidebar 
and for visbile.sidebar it would be Expand/Collapse Sidebar... 

something like it. 

 
 
--

--

const NoSSRKedro = dynamic(() => import('@quantumblack/kedro-viz'), {
  ssr: false,
});

const MyApp = () => <NoSSRKedro data={json} />;
```

The JSON can be obtained by running:

```bash
kedro viz run --save-file=<path-to-save-kedro-viz-data-to-a-directory>
```

On successful execution of the command above, it will generate a folder at the specified directory, containing the following structure: 

```
<filename>/api/
├── main
├── nodes
│   ├── 23c94afb
│   ├── 28754fab
│   ├── 2ab3579f
│   ├── 329e963c
│   ├── 369acf98
│   └── ...
└── pipelines
    ├── __default__
    ├── data_processing
    ├── data_science
    └── ...
```

Use the `main` file as the input JSON for the `data` prop in your Kedro-Viz component.

We also recommend wrapping the `Kedro-Viz` component with a parent HTML/JSX element that has a specified height (as seen in the above example) in order for Kedro-Viz to be styled properly.

**_Our documentation contains [additional examples on how to visualise with Kedro-Viz.](https://docs.kedro.org/en/stable/visualisation/index.html)_**

## Configure Kedro-viz with `options`

The example below demonstrates how to configure your kedro-viz using different `options`. 

```

> <KedroViz

Reviewer 2: Hi @jitu5, as discussed we might need to update this section. (FYI)

});

const MyApp = () => <NoSSRKedro data={json} />;
```

The JSON can be obtained by running:

```bash
kedro viz run --save-file=<path-to-save-kedro-viz-data-to-a-directory>
```

On successful execution of the command above, it will generate a folder at the specified directory, containing the following structure: 

```
<filename>/api/
├── main
├── nodes
│   ├── 23c94afb
│   ├── 28754fab
│   ├── 2ab3579f
│   ├── 329e963c
│   ├── 369acf98
│   └── ...
└── pipelines
    ├── __default__
    ├── data_processing
    ├── data_science
    └── ...
```

Use the `main` file as the input JSON for the `data` prop in your Kedro-Viz component.

We also recommend wrapping the `Kedro-Viz` component with a parent HTML/JSX element that has a specified height (as seen in the above example) in order for Kedro-Viz to be styled properly.

**_Our documentation contains [additional examples on how to visualise with Kedro-Viz.](https://docs.kedro.org/en/stable/visualisation/index.html)_**

## Configure Kedro-viz with `options`

The example below demonstrates how to configure your kedro-viz using different `options`. 

```
<KedroViz
    data={json}
    options={
      expandAllPipelines: false,

>       display: {

Reviewer 2: [Nit] If we can arrange these in alphabetical order it would be nice in the docs. Same goes for the table below under each section if we can have an order.
